include/extractor/scripting_environment.hpp

```diff
@@ -65,8 +77,14 @@ class ScriptingEnvironment
                     tbb::concurrent_vector<std::pair<std::size_t, ExtractionWay>> &resulting_ways,
                     tbb::concurrent_vector<boost::optional<InputRestrictionContainer>>
                         &resulting_restrictions) = 0;
+
+    static location_handler_sparse_type *location_sparse_handler;
+    static location_handler_dense_type *location_dense_handler;
+    static void setSparseCache(location_handler_sparse_type &cache) {
+        location_sparse_handler = &cache;};
+    static void setDenseCache(location_handler_dense_type &cache) {
+        location_dense_handler = &cache;};
```

Comment:
You could do the following:
- store the cache in the extractor
- pass it into the scripting environment (which both fills it and also can use it)
- re-use the cache in the extraction containers where we're storing lon, lat coordinates anyway

The last point would make this pull request feasible to integrate since otherwise we duplicate the data which is _a lot_ on planet scale.